File: bindings/wasm/examples/src/resolve_did.js
// Copyright 2020-2022 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import {Resolver} from '@iota/identity-wasm';
import {createIdentity} from "./create_did";

/**
 A short example to show how to resolve a DID. This returns the latest DID Document.

 @param {{network: Network, explorer: ExplorerUrl}} clientConfig
 **/
async function resolveIdentity(clientConfig) {

    // Creates a new identity (see "create_did" example)
    const {key, doc, receipt} = await createIdentity(clientConfig);
    const doc_did = doc.id();

    // Retrieve the published DID Document from the Tangle.
    const resolver = await Resolver
        .builder()
        .clientConfig({
            network: clientConfig.network
        })
        .build();

    return await resolver.resolve(doc_did);
}

export {resolveIdentity};

File: examples/low-level-api/resolve_did.rs
// Copyright 2020-2021 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

//! A basic example that generates a DID Document, publishes it to the Tangle,
//! and retrieves information through DID Document resolution/dereferencing.
//!
//! See also https://www.w3.org/TR/did-core/#did-resolution and https://www.w3.org/TR/did-core/#did-url-dereferencing
//!
//! cargo run --example resolution

use identity::iota::Receipt;
use identity::iota::ResolvedIotaDocument;
use identity::iota::Resolver;
use identity::iota_core::IotaDID;
use identity::prelude::*;

mod create_did;

#[tokio::main]
async fn main() -> Result<()> {
  // Create a signed DID Document and KeyPair (see create_did.rs).
  let (document, _, _): (IotaDocument, KeyPair, Receipt) = create_did::run().await?;

  // ===========================================================================
  // Resolve DID Document
  // ===========================================================================

  let doc_did: &IotaDID = document.id();

  // Retrieve the published DID Document from the Tangle.
  let resolver: Resolver = Resolver::new().await?;
  let resolved_did_document: ResolvedIotaDocument = resolver.resolve(doc_did).await?;

  println!("Resolved DID Document > {:#?}", resolved_did_document);

  // The resolved Document should be the same as what we published.
  assert_eq!(resolved_did_document.document, document);

  Ok(())
}
